packages/models-library/src/models_library/basic_regex.py

```diff
@@ -0,0 +1,57 @@
+""" Regular expressions patterns to build pydantic contrained strings
+
+    - Variants of the patterns with 'Named Groups' captured are suffixed with NG_RE
+"""
+# TODO: for every pattern we should have a formatter function
+# NOTE: some sites to manualy check ideas
+#   https://regex101.com/
+#   https://pythex.org/
+#
+
+# Universally unique Identifier. Pattern taken from https://stackoverflow.com/questions/136505/searching-for-uuids-in-text-with-regex
+UUID_RE = r"\b[0-9a-f]{8}\b-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-\b[0-9a-f]{12}\b"
+
+# Formatted timestamps with date and time
+DATE_RE = r"\d{4}-(12|11|10|0?[1-9])-(31|30|[0-2]?\d)T(2[0-3]|1\d|0?[0-9])(:(\d|[0-5]\d)){2}(\.\d{3})?Z"
+
+
+## VERSIONS --------------------------------------------------------------------------------------------------
+
+VERSION_RE = r"^(0|[1-9]\d*)(\.(0|[1-9]\d*)){2}(-(0|[1-9]\d*|\d*[-a-zA-Z][-\da-zA-Z]*)(\.(0|[1-9]\d*|\d*[-a-zA-Z][-\da-zA-Z]*))*)?(\+[-\da-zA-Z]+(\.[-\da-zA-Z-]+)*)?$"
+
+# Semantic version regex (https://semver.org/)
+SEMANTIC_VERSION_RE = r"^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$"
+SEMANTIC_VERSION_NG_RE = r"^(?P<major>0|[1-9]\d*)\.(?P<minor>0|[1-9]\d*)\.(?P<patch>0|[1-9]\d*)(?:-(?P<prerelease>(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+(?P<buildmetadata>[0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$"
+
+# PEP 440 -- Version Identification and Dependency Specification (https://www.python.org/dev/peps/pep-0440)
+PEP404_VERSION_RE = r"^([1-9][0-9]*!)?(0|[1-9][0-9]*)(\.(0|[1-9][0-9]*))*((a|b|rc)(0|[1-9][0-9]*))?(\.post(0|[1-9][0-9]*))?(\.dev(0|[1-9][0-9]*))?$"
+PEP440_VERSION_NG_RE = r"""
```

Comment:
as I commented. I would keep with VERSION_RE unless we have a valid case where we would need something more advanced and define when and how we use that.

Reply:
Fine with me. Nonetheless, i might need those in the new service integration library we are building